### ISSUE TITLE
Passing writer_options with generate()

### ISSUE THREAD

**Issue by ildarakhmetov** [WhyNotHugo/python-barcode#38]
I'm calling `generate()` function to create a code128 barcode, and I need to pass custom `writer_options` payload. Here is the code fragment:

```
writer_options = {
    'module_width': 0.3,
    'module_height': 25.0,
    'quiet_zone': 1,
    'font_size': 10,
    'text_distance': 0,
    'background': 'white',
    'foreground': 'black',
    'write_text': True,
    'text': '',
}
name = generate('code128', foo, output=f"barcode_{foo}", text=' ', writer=ImageWriter(), writer_options=writer_options)
```

This code gives me the following error message:
```
Traceback (most recent call last):
  File "gen_barcode_foreign.py", line 55, in <module>
    name = generate('code128', tracking, output=f"barcode_{tracking}", text=' ', writer=ImageWriter(), writer_options=writer_options)
  File "d:\python_test\barcodes\venv\lib\site-packages\barcode\__init__.py", line 73, in generate
    barcode = get(name, code, writer, options)
  File "d:\python_test\barcodes\venv\lib\site-packages\barcode\__init__.py", line 61, in get
    return barcode(code, writer, **options)
TypeError: __init__() got an unexpected keyword argument 'module_width'
```

I tried to edit `__init__.py`, changing the line #61 from `return barcode(code, writer, **options)` to `return barcode(code, writer)`, and it helped, `generate()` now works fine.

Not sure if this is the correct way to solve this issue though.

**Comment by brenramz** [comment]
I found that error too and I realized the init of Code128 is missing **kwargs param (file codex.py line 139)

**Comment by adast** [comment]
Better solution:
With this solution you can set up options for writer and code at the same time
`def generate(name, code, writer=None, output=None, writer_options=None, code_options=None,
             text=None, pil=False):
    barcode = get(name, code, writer, code_options)
    if pil:
        return barcode.render(writer_options, text)
    if isinstance(output, _strbase):
        fullname = barcode.save(output, code_options, text)
        return fullname
    else:
        barcode.write(output, code_options, text)`

Usage:
from barcode.writer import ImageWriter
from barcode import generate
generate(
    'CODE39',
    '130000123',
    pil=True,
    writer=ImageWriter(),
    writer_options={'module_height': 5.0, 'text_distance': 1.0},
    code_options={'add_checksum': False}
)

**Comment by WhyNotHugo** [comment]
This is now fixed by @987da3b696050e04a9a3555693cf421a0c3ee146.
